src/app/component/NotificationList/NotificationList.component.js

```diff
@@ -14,33 +14,50 @@ import PropTypes from 'prop-types';
 import Notification from 'Component/Notification';
 import { NotificationListType } from 'Type/NotificationList';
 import './NotificationList.style';
+import { DENIED } from 'Store/Notification/Notification.reducer';
+import PushNotification from 'Component/PushNotification';
 
 /**
  * Notification List
  * @class NotificationList
  */
 class NotificationList extends Component {
     render() {
-        const { onHideNotification, notifications } = this.props;
+        const {
+            onHideNotification,
+            notifications,
+            grantType,
+            supported
+        } = this.props;
+        const sw = Boolean(navigator.serviceWorker);
 
         return (
             <div className="NotificationList">
-                { Object.keys(notifications).map(id => (
-                    <Notification
-                      key={ id }
-                      notificationId={ id }
-                      notification={ notifications[id] }
-                      onHideNotification={ onHideNotification }
-                    />
-                )) }
+                { Object.keys(notifications).map((id) => {
+                    return supported && grantType !== DENIED && sw ? (
```

Comment:
I assume this could raise issues. Could you please add a `()` around logical operators. I might be wrong here, just telling from experience...